### ISSUE TITLE
Non-default transport compatibility

### ISSUE THREAD

**Issue by davidjytang** [jdthorpe/winston-log-and-exit#2]
Thank you for creating this package.

Have you had any success with non-default tranport?
Currently I'm trying to log to syslog with https://www.npmjs.com/package/winston-rsyslog2
Console transported logs are directed to a file correctly after starting to use log_and_exit().
However syslog are not receiving.

Would you mind sharing your experience on solving this exit issue with syslog using winston?

**Comment by jdthorpe** [comment]
Hi David,

I haven't thought about this in a long time, but it seems that the issue is that the rsyslog2 does not implement as streaming interface which is required in order to work with winston log-and-exit.  This is the relevent line in winston log-and-exit:

https://github.com/jdthorpe/winston-log-and-exit/blob/master/index.js#L13

which checks if the transport has a `_stream` attribute, and likewise it doesn't issue a "finish" event, which would be required in order for `winston-log-and-exit` to be effective. . Unfortunately the streaming API may not be well documented, and I had to glean it from looking at the internals of winston (which is probably why this was / continues to be a hard problem for people to solve).

I opened an [issue with rsyslog2](https://github.com/imsky/winston-rsyslog2/issues/9) and added the [required streaming transport API to my README](https://github.com/jdthorpe/winston-log-and-exit/blob/master/README.md#streaming-api-for-custom-transports).

Cheers,

-Jason
